### ISSUE TITLE
[Bug]: setFullScreen not working in conjunction with setFullScreenable

### ISSUE THREAD

**Issue by jtvberg** [electron/electron#37852]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

23.1.4

### What operating system are you using?

macOS

### Operating System Version

macOS Monterey 12.6.3

### What arch are you using?

x64

### Last Known Working Electron version

11.4.3

### Expected Behavior

Using `setFullScreen(false)` should take the window out of full screen.

### Actual Behavior

When using `setFullScreen(false)` in conjunction with `setFullScreenable(false)` the window does not exit from full screen.

Even in this order, it doesn't work.
```
mainWindow.setFullScreen(false)
mainWindow.setFullScreenable(false)
```

### Testcase Gist URL

https://gist.github.com/jtvberg/b3dde9d24bb8e924aa0e726f77392581

### Additional Information

Works as expected in 11.4.3 and stops working in 11.4.4
Looks like it is associated with this: https://github.com/electron/electron/pull/28773

A work-around is commented out in the gist.
Commenting out
`mainWindow.setFullScreenable(false)`
and then using the second timeout in the comment makes things work as intended.

The initial timeout is just to avoid all the other code to create a UI.

**Comment by jtvberg** [comment]
@codebytere no worries. I figured as much and I have a work around. 
I just couldn't find anyhing on this so I figured at least there would be an issue entry in case someone else ends up banging their head on this one.

It would probably be a good idea for the [wiki](https://www.electronjs.org/docs/latest/api/browser-window#winsetfullscreenflag) entry to note the async nature of the setFullScreen call.
